wasm/validator/derive/src/token.rs

```diff
@@ -26,17 +26,21 @@ fn impl_token(ident: &syn::Ident, generics: &syn::Generics) -> proc_macro2::Toke
     quote! {
         impl #impl_generics ::iroha_validator::permission::Token for #ident #ty_generics #where_clause {
             fn is_owned_by(&self, account_id: &::iroha_validator::data_model::prelude::AccountId) -> bool {
-                let all_account_tokens: Vec<PermissionToken> = ::iroha_validator::iroha_wasm::debug::DebugExpectExt::dbg_expect(
-                    ::iroha_validator::iroha_wasm::QueryHost::execute(
-                        &::iroha_validator::iroha_wasm::data_model::prelude::FindPermissionTokensByAccountId::new(
+                let account_tokens_cursor = ::iroha_validator::iroha_wasm::debug::DebugExpectExt::dbg_expect(
+                    ::iroha_validator::iroha_wasm::ExecuteQueryOnHost::execute(
+                        ::iroha_validator::iroha_wasm::data_model::prelude::FindPermissionTokensByAccountId::new(
                             account_id.clone(),
                         )
                     ),
                     "Failed to execute `FindPermissionTokensByAccountId` query"
-                ).try_into().unwrap();
+                );
 
-                all_account_tokens
+                account_tokens_cursor
                     .into_iter()
+                    .map(|res| ::iroha_validator::iroha_wasm::debug::DebugExpectExt::dbg_expect(
```

Comment:
```suggestion
                account_tokens_cursor
                    .map(|res| ::iroha_validator::iroha_wasm::debug::DebugExpectExt::dbg_expect(
```

isn't it already an iterator?

Reply:
Well, no, it's a type which has some other functions like `into_inner()` and `collect()`. It implements `into_iter()` only if its generic is `Vec`.

From the other side I think I could make it to be an iterator it-self, but then I'm not sure about functions I wrote above. Especialy `collect()` which has a bit another meaning than `Iterator::collect()`.